.github/workflows/main.yml

```diff
@@ -89,6 +89,17 @@ jobs:
             cd taco_parser
             cargo build --release
 
+      - name: Install protoc
+        run: sudo apt-get install protobuf-compiler 
+
+      - name: Build xml_converter
+        run: |
+            cd xml_converter
+            protoc --proto_path=generators/proto_templates --cpp_out=src/ generators/proto_templates/waypoint.proto
```

Comment:
This works as intended but I'm curious if there is a best practice. Since the Makefile for xml_convertor is not included in the code, cmake needs to be run. Therefore I thought it'd be best for the proto buffers to be generated here. Similarly I could add a line to run code_generator.py

Reply:
This is the equivalent of "please manually run `protoc`" which is not a good build experience. Protoc should be run inside the generated makefile as originally requested.

In general there is no issue with putting commands in the CI script, however in this  particular case it should be a part of the makefile. Right now if you try to type `cmake . && make` it would crash on compilation because the generated proto files would not exist.